### ISSUE TITLE
azurerm_cosmosdb_mongo_collection does not support having shard_key = null when specifying autoscale_settings

### ISSUE THREAD

**Issue by Raffledoocious** [hashicorp/terraform-provider-azurerm#15495]
I am trying to import existing Mongo CosmosDB collections which have autoscaling enabled but are _not_ sharded. However, terraform requires the "shard_key" field when specifying autoscale settings. I am not able to match the configuration of my Mongo collections in Azure and can't map the resources.

```
resource "azurerm_cosmosdb_mongo_collection" "test" {
  name                = "test"
  resource_group_name = local.rg_name
  account_name        = local.cosmosdb_account_name
  database_name       = azurerm_cosmosdb_mongo_database.test_db.name

  autoscale_settings {
    max_throughput = 4000
  }

  index {
    keys   = ["_id"]
    unique = true
  }

  index {
    keys   = ["$**"]
    unique = false
  }
}

```

```
╷
│ Error: Missing required argument
│ 
│   with azurerm_cosmosdb_mongo_collection.test,
│   on cosmos_db.tf line 47, in resource "azurerm_cosmosdb_mongo_collection" "test":
│   47: resource "azurerm_cosmosdb_mongo_collection" "test" {
│ 
│ "autoscale_settings": all of `autoscale_settings,shard_key` must be
│ specified
╵
```

This should be a supported configuration because the collection in Azure is specified as not having a shard. Here is the same resource returned from the Azure CLI. Notice the _shardKey_ property is null.

```
{
  "id": "/subscriptions/a7a2514f-1b96-4dee-84c6-7e6056ba5097/resourceGroups/test-rg-001/providers/Microsoft.DocumentDB/databaseAccounts/test-rg-001/mongodbDatabases/test/collections/test",
  "location": null,
  "name": "test",
  "options": null,
  "resource": {
    "analyticalStorageTtl": null,
    "etag": null,
    "id": "test",
    "indexes": [
      {
        "key": {
          "keys": [
            "_id"
          ]
        },
        "options": null
      },
      {
        "key": {
          "keys": [
            "$**"
          ]
        },
        "options": null
      }
    ],
    "rid": null,
    "shardKey": null,
    "ts": null
  },
  "resourceGroup": "test-rg-001",
  "tags": null,
  "type": "Microsoft.DocumentDB/databaseAccounts/mongodbDatabases/collections"
}
```

Creating autoscaling collections this way is supported in the Azure portal as well. 
![image](https://user-images.githubusercontent.com/1238728/154591992-4ad256a6-740d-4998-b1f8-2a28f403450d.png)

Is this a legitimate bug? If so I could look at what it would take to rectify it and submit a PR.

Terraform version - 1.05
azurerm module version - 2.81.0

**Comment by sinbai** [comment]
@Raffledoocious thank you for opening this issue. I have submitted PR #15529 to fix this issue.

**Comment by Raffledoocious** [comment]
Great to hear and thank you for the PR. I will keep tabs on this and of course if you need me to test anything don't hesitate to reach out.

**Comment by sinbai** [comment]
@Raffledoocious That fix is expected to be released in the upcoming release 2.98, so please track that to see whether you would then be unblocked, let me know if you have any questions.
